src/PIL/GimpPaletteFile.py

```diff
@@ -24,33 +25,49 @@ class GimpPaletteFile:
 
     rawmode = "RGB"
 
-    def __init__(self, fp):
+    #: override if reading larger palettes is needed
+    max_colors = 256
+    _max_line_size = 100
+    _max_file_size = 2**20
 
-        self.palette = [o8(i) * 3 for i in range(256)]
+    def __init__(self, fp):
 
         if fp.readline()[:12] != b"GIMP Palette":
             raise SyntaxError("not a GIMP palette file")
 
-        for i in range(256):
+        read = 0
 
-            s = fp.readline()
+        self.palette = b""
+        while len(self.palette) < 3 * self.max_colors:
+
+            s = fp.readline(self._max_file_size)
             if not s:
                 break
 
+            read += len(s)
+            if read >= self._max_file_size:
+                warnings.warn(
+                    f"Palette file truncated at {self._max_file_size - len(s)} bytes"
+                )
+                break
+
             # skip fields and comment lines
             if re.match(rb"\w+:|#", s):
                 continue
-            if len(s) > 100:
+            if len(s) > self._max_line_size:
                 raise SyntaxError("bad palette file")
 
-            v = tuple(map(int, s.split()[:3]))
-            if len(v) != 3:
+            # 4th column is color name and may contain spaces.
+            v = s.split(None, 4)
+            if len(v) < 3:
                 raise ValueError("bad palette entry")
-
-            self.palette[i] = o8(v[0]) + o8(v[1]) + o8(v[2])
-
-        self.palette = b"".join(self.palette)
+            for i in range(3):
+                self.palette += o8(int(v[i]))
 
     def getpalette(self):
 
         return self.palette, self.rawmode
+
+    @property
+    def n_colors(self):
+        return len(self.palette) / 3
```

Comment:
[As I've said](https://github.com/jsbueno/Pillow/pull/1#issue-1398563514), i don't think that `n_colors` is necessary. If the user is going to use the palette string, then they have enough of an understanding to be able to determine how many colors there are. This is a simple operation without an obvious need, so I'd like to avoid adding to the API for a trivial case.

But if you disagree, you can leave it here and see what others think.

Reply:
I will remove it. However, I definetelly like Python things to be "pythonic".  That include bring able to know about object states through properties, usingagic.methods so users can comfortably use operators and common interfaces to deal with their data.

In my experience these things make libraries more joyful to use and easier to maintain.

Pillow Is a project that tends to the opposite: a lot of functional things and magic global states. I think part of that is due to historic reasons than a philosophy of the project, and hope to offer more pythonic interfaces like I had indicated in the chat about animared images.

That said, I'd say these comfortable properties could be left for a revamp of the ImagPallete class ata future time, and there is no need for the new property here.

This will also give time for all involved to understand and have an opinion if this "first class classes" approach is desirable at all, and draw, projectwise, a more solid roadmap.


(As far as I am concerned wanting to know the number of colors of a palette is an operation common enough and with nothing of unexpected - akim to getting a strong length - and requiring one to deal with bytes when their problem is dealing with colors is a quasi definition of "unpythonic". However, if this class is to be instrumented,.maybe this could be offered as it's `__len__` for once, instead of creating an arbitrarily named property.)